palette/build/named.rs

```diff
@@ -40,7 +47,56 @@ pub fn build() {
         entries.push((name.to_owned(), name.to_uppercase()));
     }
 
-    gen_from_str(&mut writer, &entries)
+    gen_from_str(writer, &entries)
+}
+
+#[cfg(feature = "named_gradients")]
+pub fn build_gradients(writer: &mut File) {
+    use std::io::{BufRead, BufReader, Write};
+
+    let reader =
+        BufReader::new(File::open("build/svg_gradients_mpl.txt").expect("could not open svg_gradients_mpl.txt"));
+
+    let mut line_iter = reader.lines();
+    while let Some(Ok(line)) = line_iter.next(){
+        //empty lines are allowed
+        if line.is_empty() {continue;}
+        let mut parts = line.split_whitespace();
+        //every line should have the same info: name type number_of_colors [\n red green blue]^number_of_colors
+        let name = parts.next().expect("couldn't get the color name");
+        let color_type = parts.next().expect("couldn't get the type of the colors");
+        //we assume that color_type is a rgb type
+        let color_type = format!("crate::rgb::{}", color_type);
+        let number_of_colors : usize = parts.next().expect("couldn't get the number of colors")
+            .parse().unwrap_or_else(|_| panic!("couldn't parse the number of colors for color {}", name));
+        writeln!(writer, "/// New matplotlib colormap by Nathaniel J. Smith, Stefan van der Walt, and (in the case of viridis) Eric Firing.").unwrap();
+        writeln!(writer, "/// This gradient is perfectly perceptually-uniform, both in regular form and also when converted to black-and-white.").unwrap();
+        //writeln!(writer, "/// The colormap is released under the CC0 license public domain dedication.").unwrap();
```

Comment:
This should probably not be commented out. We can also link to an official source if possible.

Reply:
I commented it out because I tried to be in line with the named colors, which do not include any license in the documentation of each color but do include the source in the documentation of the named module. Such I also included the source in the documentation of the `gradient/named` module.

How and where to include credits and licenses if of course up to you. So my questions are these:
- What should I include in the documentation of the named module?
- What should I include in the documentation of each gradient?

Things possible to include would be:
- License
- Content of the license
- Source
- Authors / Creators

Reply:
Welcome to software licensing. 😬 I think you can just uncomment the line and it will be fine. We are then mentioning the authors and the license, and covering our backs even though it's public domain. Here are the guidelines for a CC license that forces attribution: https://creativecommons.org/use-remix/attribution/ Public domain doesn't require that or anything else as I understand it.